File: Append_and_Delete/python/AppendAndDelete.py
import sys

# Complete the appendAndDelete function below.
def appendAndDelete(s, t, k):
   
    getMin = (min(len(s), len(t)))
    
    firstDiffPosition = getMin
    
    for x in range(getMin):
        if (s[x] != t[x]):
            firstDiffPosition = x
            break
    steps = len(s) + len(t) - 2 * firstDiffPosition
    
    if  (k>= steps and (k-steps)%2 ==0) or k>= len(s)+len(t) or k == steps:
        return "Yes"
    else:
        return "No"

File: 2_Append_and_Delete/node/AppendAndDelete.js
function appendAndDelete(s, t, k) {
    let getMin = Math.min.apply(Math,[s.length, t.length]);
    let firstDiffPosition = getMin;

    for (let i = 0; i < getMin; i++) {
        if(s[i] != t[i]){
            firstDiffPosition = i
            break;
        }
    }
    let steps = s.length + t.length - 2 * firstDiffPosition;
    
    if((k>=steps && (k-steps)%2==0) ||  (k>= s.length + t.length) || (k == steps)) {
        return "Yes";
    }
    else {
        return "No";
    }

}